components/auth/org.wso2.carbon.auth.scim.rest.api/src/main/java/org/wso2/carbon/auth/scim/rest/api/impl/UsersApiServiceImpl.java

```diff
@@ -1,57 +1,107 @@
 package org.wso2.carbon.auth.scim.rest.api.impl;
 
-import org.wso2.carbon.auth.scim.rest.api.*;
-import org.wso2.carbon.auth.scim.rest.api.dto.*;
-
-
-import java.util.List;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.auth.core.exception.AuthUserManagementException;
+import org.wso2.carbon.auth.scim.impl.SCIMManager;
 import org.wso2.carbon.auth.scim.rest.api.NotFoundException;
-
-import java.io.InputStream;
-
-import org.wso2.msf4j.formparam.FormDataParam;
-import org.wso2.msf4j.formparam.FileInfo;
+import org.wso2.carbon.auth.scim.rest.api.UsersApiService;
+import org.wso2.carbon.auth.scim.rest.api.dto.UserDTO;
+import org.wso2.carbon.auth.scim.rest.api.dto.UserSearchDTO;
+import org.wso2.carbon.auth.scim.rest.api.util.ApiServiceUtils;
+import org.wso2.charon3.core.extensions.UserManager;
+import org.wso2.charon3.core.protocol.SCIMResponse;
+import org.wso2.charon3.core.protocol.endpoints.UserResourceManager;
 import org.wso2.msf4j.Request;
+
 import javax.ws.rs.core.Response;
-import javax.ws.rs.core.SecurityContext;
 
 public class UsersApiServiceImpl extends UsersApiService {
+    private static final Log LOG = LogFactory.getLog(UsersApiServiceImpl.class);
+
     @Override
-    public Response usersGet(Integer startIndex
-, Integer count
-, String filter
-  ,Request request) throws NotFoundException {
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+    public Response usersGet(Integer startIndex, Integer count, String filter, Request request)
+                                                                                    throws NotFoundException {
+        UserManager userManager;
+        try {
+            userManager = SCIMManager.getInstance().getCarbonAuthUserManager();
+            UserResourceManager userResourceManager = new UserResourceManager();
+            SCIMResponse scimResponse = userResourceManager.listWithGET(userManager, filter, startIndex, count,
+                    null, null, null, null);
+            return ApiServiceUtils.buildResponse(scimResponse);
+        } catch (AuthUserManagementException e) {
+            LOG.error("Error in initializing the CarbonAuthUserManager");
```

Comment:
Exception has not been logged. Need to log exception as follows,

log.error("Message", e);